src/lib/genesis_proof/genesis_proof.ml

```diff
@@ -85,6 +86,7 @@ module T = struct
     ; constraint_constants : Genesis_constants.Constraint_constants.t
     ; genesis_constants : Genesis_constants.t
     ; proof_level : Genesis_constants.Proof_level.t
+    ; compile_config : Mina_compile_config.t
```

Comment:
This field is added so that it later appears in `Precomputed_values.t` and in all of the places that get it.

I don't like this solution. I tried to revert it on `compatible`, but there turned out to be many places which need to be updated.

I suggest to leave it "as is" and later find a better solution for passing `Mina_compile_config.t` to various structures. BTW, the name will soon be completely pointless, as this config is configured in runtime now as much as in runtime. I suggest after refactoring to get rid of any compile-time configuration whatsoever we reconsider use of both `Mina_compile_config.t` and `Precomputed_values.t`. And skip spending time on partial solutions now